src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+        int numPLHP = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+        if (numPLHP > 0) {
+            auto const instances = state.dataInputProcessing->inputProcessor->epJSON.find(cCurrentModuleObject);
+            if (instances == state.dataInputProcessing->inputProcessor->epJSON.end()) {
+                // Cannot imagine how you would have numPLHP > 0 and yet the instances is empty
+                // this would indicate a major problem in the input processor, not a problem here
+                // I'll still catch this with errorsFound but I cannot make a unit test for it so excluding the line from coverage
+                ShowSevereError(state,                                                                     // LCOV_EXCL_LINE
+                                "EIR PLFFHP: Somehow getNumObjectsFound was > 0 but epJSON.find found 0"); // LCOV_EXCL_LINE
+                errorsFound = true;                                                                        // LCOV_EXCL_LINE
+            }
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = instance.key();
+                state.dataInputProcessing->inputProcessor->markObjectAsUsed(cCurrentModuleObject, thisObjectName);
+
+                EIRFuelFiredHeatPump thisPLHP;
+                thisPLHP.EIRHPType = classToInput.thisType;
+                // A1-A3
+                thisPLHP.name = UtilityRoutines::MakeUPPERCase(thisObjectName);
+                std::string loadSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_inlet_node_name").get<std::string>());
+                std::string loadSideOutletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_outlet_node_name").get<std::string>());
+                // Implicit
+                std::string condenserType = "AIRSOURCE"; // UtilityRoutines::MakeUPPERCase(fields.at("condenser_type").get<std::string>());
+                // A4
+                std::string sourceSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("air_source_node_name").get<std::string>());
+                std::string sourceSideOutletNodeName =
+                    "DUMMY_CONDENSER"; // UtilityRoutines::MakeUPPERCase(fields.at("source_side_outlet_node_name").get<std::string>());
+                srand(time(NULL));
+                for (int i = 1; i <= 3; ++i) {
+                    sourceSideOutletNodeName += ("_" + std::to_string(rand()));
+                }
+
+                // A5
+                if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) {
+                    if (fields.find("companion_cooling_heat_pump_name") != fields.end()) { // optional field
```

Comment:
This is not a good pattern because you are searching for the same string in the map twice, once in `find` and once in `at`. A better pattern is:

```cpp
auto found = fields.find("companion_cooling_heat_pump_name");
if (found != fields.end()) 
   thisPLHP.companionCoilName = MakeUPPERCase(found.get<std::string>()); // I think epJSON uppercases things for you already.
```

Reply:
I was trying to follow this before and the compiler will not let pas the `found.get<std::string>()` part. I think from the newer comments this morning that it need to be `found.value().get<std::string>()` instead, which will then work ok.

Reply:
These were all addressed in previous changes.